Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m

```diff
@@ -510,6 +512,9 @@ void FIRPopulateProtoWithInstalledServices(logs_proto_mobilesdk_ios_ICoreConfigu
   logs_proto_mobilesdk_ios_ICoreConfiguration_ServiceType *servicesInstalled =
       malloc(sizeof(logs_proto_mobilesdk_ios_ICoreConfiguration_ServiceType) *
              sdkServiceInstalledArray.count);
+  if (servicesInstalled == NULL) {
+    return servicesInstalled;
```

Comment:
tiniest of nits: It may be simpler here to explicitly return `logs_proto_mobilesdk_ios_ICoreConfiguration_ServiceType_UNKNOWN_SDK_SERVICE` instead of `servicesInstalled`.

Reply:
The failure isn't because the sdk service is unknown, it's because we couldn't allocate memory to serialize the sdkService array to a proto in memory. Also, I messed up and there shouldn't have been  a return value.